src/sentry/scim/endpoints/teams.py

```diff
@@ -192,6 +192,13 @@ def post(self, request: Request, organization) -> Response:
         request.data.update(
             {"name": request.data["displayName"], "slug": slugify(request.data["displayName"])}
         ),
+
+        if "scim" in request.user.username:
+            request.user.username = (
+                "SCIM Okta Integration (" + request.user.username.split("-")[3] + ")"
```

Comment:
`username.split("-")[3]` is a little cryptic. I would introduce a small utility function for unpacking a SCIM integration's username. Ideally you would unpack the parts into a tuple that explains what they are, such as:
```python
prefix, something, name, suffix = request.user.username.split("-")
```
(with the variable names replaced, of course).